tests/integration/__init__.py

```diff
@@ -97,6 +98,12 @@ def get_server_versions():
     return (cass_version, cql_version)
 
 
+def get_scylla_version(scylla_ccm_version_string):
+    """ get scylla version from ccm before starting a cluster"""
+    ccm_repo_cache_dir, _ = ccmlib.scylla_repository.setup(version=scylla_ccm_version_string)
```

Comment:
What actions does `setup` perform? Because it sounds like it actually creates a new cluster (maybewithout starting it, idk),  which is quite a lot of work to just get a version.
Why is this PR necessary? Did CI fail because of this, is there some issue opened?

Reply:
setup is making sure the version is downloaded and in place

It's failing at any time you don't use release:5.4

We are testing daily builds in the matrix, it was for quite some time a patch there.

Again the assumption test should only work correctly with released versions only us wrong

Reply:
Are the actions performed by `setup` performed only once when it is called multiple times?
So if we call `get_scylla_version` and then some other place in the code calls `setup` will all this work be performed once or twice?

Reply:
If the work is performed only once per argument then it is fine imo, no need to add any cache